src/main/java/dev/jbang/cli/Run.java

```diff
@@ -210,42 +209,44 @@ String generateCommandLine(Script script) throws IOException {
 						if (agent.getJar() != null) {
 							jar = agent.getJar().toString();
 						} else if (agent.forJar()) {
-							jar = agent.getBackingFile().toString();
+							jar = agent.getResourceRef().getFile().toString();
 							// should we log a warning/error if agent jar not present ?
 						}
 						if (jar == null) {
 							throw new ExitException(EXIT_INTERNAL_ERROR,
-									"No jar found for agent " + agent.getOriginalRef());
+									"No jar found for agent " + agent.getResourceRef().getOriginalResource());
 						}
 						fullArgs.add("-javaagent:" + jar
 								+ (agent.getJavaAgentOption() != null ? "=" + agent.getJavaAgentOption() : ""));
 
 					});
 
-			fullArgs.addAll(script.collectRuntimeOptions());
-			fullArgs.addAll(script.getAutoDetectedModuleArguments(requestedJavaVersion, offline));
+			if (xrunit.runUnit instanceof Script) {
```

Comment:
this is not only scripts - jars could have runtime options....but yeah right now it won't.

Reply:
Sure, but `collectRuntimeOptions()` is only for `//JAVA_OPTS` which only exists on scripts.

Reply:
Technically you can specify via env vars too.